evalai/utils/requests.py

```diff
@@ -11,75 +10,51 @@
 
 
 def make_request(path, method, files=None, data=None):
+    if method in ["PUT", "PATCH", "DELETE"]:
+        raise ValueError("Unsupported method: {}".format(method))
+
     url = "{}{}".format(get_host_url(), path)
     headers = get_request_header()
+    if method == "POST":
+        files = {"input_file": open(files, "rb")} if files else None
+        data = {"status": "submitting"}
 
-    if method == "GET":
-        try:
-            response = requests.get(url, headers=headers)
-            response.raise_for_status()
-        except requests.exceptions.HTTPError as err:
-            if response.status_code in EVALAI_ERROR_CODES:
-                validate_token(response.json())
-                echo(
-                    style(
-                        "\nError: {}\n".format(response.json().get("error")),
-                        fg="red",
-                        bold=True,
-                    )
-                )
-            else:
-                echo(err)
-            sys.exit(1)
-        except requests.exceptions.RequestException:
+    try:
+        response = requests.request(method, url, data=data, headers=headers, files=files)
+        response.raise_for_status()
+    except requests.exceptions.HTTPError as e:
+        if response.status_code in EVALAI_ERROR_CODES:
+            validate_token(response.json())
             echo(
                 style(
-                    "\nCould not establish a connection to EvalAI."
-                    " Please check the Host URL.\n",
-                    bold=True,
+                    "\nError: {}\n".format(response.json().get("error")),
```

Comment:
@nikochiko can we add some description of the error like it was there before `Could not establish a connection to EvalAI`

Reply:
@Ram81 That error message is more appropriate for the RequestException where it is being shown, see [this line](https://github.com/Cloud-CV/evalai-cli/pull/237/files#diff-e7df4381b09aa49cce42fde57be6249dR49). As this handling part is for HTTP errors, the error types may vary and therefore I have left this generalized message.

Reply:
@Ram81 @nikochiko 
How about
```
Could not establish a connection to EvalAI with error: {ERROR}
```

Reply:
@vkartik97 But it could also be a HTTP-500 error / HTTP-40X error when the connection was established but the request couldn't be completed. The reference to a connection issue could be misleading in such a case.

Reply:
@Ram81 I have not made any changes in the behavior of this function, I think the GitHub differences notation is a bit misleading, the error message being shown in red was for the handling of `requests.exceptions.RequestException` and the one in green that is being handled here is for HTTPError.

Reply:
Pardon me, I didnt explain better.
I meant that 
```
except requests.exceptions.HTTPError as e:
.
.
.
.
          echo("Could not establish a connection to EvalAI with error: {0}".format(e))
```
The error should give error that `requests` encountered. This will give context to the user of the type of error.

Reply:
@vkartik97 Okay, that looks good, Actually we use the same error message for `requests.exceptions.RequestException`, so I think we can merge both. We also validate_token in between if a known error is thrown by the server, So, how about this to combine:
```python
except requests.exceptions.RequestException as e:
    if isinstance(e, requests.exceptions.HTTPError) and response.status_code in EVALAI_ERROR_CODES:
        validate_token(response.json())
        e = response.json().get("error")  # In this case, the error message is returned by the server
    echo("Could not establish a connection to EvalAI with error: {}".format(e))
``` 
This will shorten and generalize the error handling. Is this good?

Reply:
I think I had misunderstood it, I will change it according to the syntax mentioned!